### PR TITLE
patch to not import networking module if on windows platform

### PR DESCRIPTION
**Pull request by snuffysasa**
let me know what you want to change, a little hacky,

it someone on windows attempts the network features it won't work and won't be very verbose on why it's not working except for the print line at the very start:
print("Running from windows platform, disabling network features")

### CODE REVIEW

**Comment by simonlindholm** [comment]
Yeah that print is ugly, let's emit it only if someone actually passes -J or similar. Also, let's make sure it's not possible to end up with crashes due to accesses to start_client, ServerError, connect, enable_debug_mode. We should get mypy passing on Windows.

**Comment by snuffysasa** [comment]
mypy is passing now with this patch. I will work on making this a cleaner

but the challenge with checking -J... is that the imports happen way before the arg parsing happens.

**Comment by snuffysasa** [comment]
but I guess on windows could just set a bool that it wasn't imported, then if someone does use -J, then it can throw the error message at that point, so I can work on that

**Comment by simonlindholm** [comment]
> mypy is passing now with this patch. I will work on making this a cleaner

then that's a bug, because clearly there are code paths that reference unset variables :) maybe it needs to be told about what initial environment to expect? I remember there being mypy docs about how to deal with platform-specific code

> but the challenge with checking -J... is that the imports happen way before the arg parsing happens.

one option would be to not do that, performing the imports within relevant functions instead

> but I guess on windows could just set a bool that it wasn't imported, then if someone does use -J, then it can throw the error message at that point, so I can work on that

sure, that works too

**Comment by snuffysasa** [comment]
>one option would be to not do that, performing the imports within relevant functions instead

You can do imports inside of functions?

**Comment by snuffysasa** [comment]
>then that's a bug, because clearly there are code paths that reference unset variables :) 

You're right, kind of strange mypy doesn't complain

**Review by simonlindholm — COMMENTED** [review]
> > one option would be to not do that, performing the imports within relevant functions instead
>
> You can do imports inside of functions?

Yes. Seems like the network functions are actually only used in one place in main.py; could we move the imports there?

File: src/main.py [hunk 1]
```diff
@@ -22,8 +22,20 @@
...
+system = platform.system().lower()
+
+isWindowsPlatform: bool = "nt" in system or "msys" in system
+
+if not isWindowsPlatform:
+    from .net.client import start_client
+    from .net.core import ServerError, connect, enable_debug_mode, MAX_PRIO, MIN_PRIO
+else:
+    MAX_PRIO = 0.0
+    MIN_PRIO = 0.0
```
[the remark this answers is not in the document]
Comment (snuffysasa) [reply src/main.py:37]: what?! you mean you don't like dancing?

File: src/main.py [hunk 2]
```diff
@@ -25,8 +25,25 @@
...
+    from .net.core import ServerError, connect, enable_debug_mode
+except ModuleNotFoundError as exception:
+    if exception.name == "nacl":
+        print(
+            "Warning: failed to load module, nacl (PyNaCl), needed for networking mode"
+        )
+        missing_nacl = True
+    else:
+        print("ModuleNotFoundError: ", exception)
+        sys.exit(1)
```
Comment (simonlindholm) [inline src/main.py:41]:
```suggestion
        `raise`
```

**Review by simonlindholm — COMMENTED** [review]
> Seems like the network functions are actually only used in one place in main.py; could we move the imports there?

Repeating this comment

**Comment by snuffysasa** [comment]
> > Seems like the network functions are actually only used in one place in main.py; could we move the imports there?
> 
> Repeating this comment

Yea I figured, it was either or.  This solution or the other.

I can move the imports to the spot they are used in main... but If I do that, then there is no longer a need for the explicit checking for nacl and a try catch....

it would just throw module not found error when someone is on window (or doesn't have nacl) and trys networking mode.    And that sounds like a fine solution to me.  (originally I didn't realize I could move imports like that)

We could have like a warning message that says "You may have trouble installing Nacl if you are on windows"... but not really necessary IMO.

In this case, I'll probably scrap this PR, and just make a clean one that does this solution.

**Pull request closed without merging**